### PR TITLE
[firejail] Add Sailfish OS privileged-data option. JB#51951

### PR DESCRIPTION
**Pull request by spiiroin**
Allows limiting visibility of SailfishOs specific privileged user data
directories.

Signed-off-by: Simo Piiroinen <simo.piiroinen@jolla.com>

### CODE REVIEW

File: rpm/0007-Implement-SailfishOS-specific-private-privileg.patch [hunk 1]
```diff
@@ -0,0 +1,668 @@
...
++	if (cfg.privileged_private_keep) {
++		if (cfg.chrootdir) {
++			fwarning("private-privileged feature is disabled in chroot\n");
++		}
++		else if (arg_overlay) {
++			fwarning("private-privileged feature is disabled in overlay\n");
++		}
++		else {
++			char *private_dir = 0;
++			if (asprintf(&private_dir, "%s/.local/share/system/privileged",cfg.homedir) < 0)
```
[the remark this answers is not in the document]
Comment (spiiroin) [reply rpm/0007-Implement-SailfishOS-specific-private-privileg.patch:443]: ack, tried to look for similar issues else where too

File: rpm/0007-Implement-SailfishOS-specific-private-privileg.patch [hunk 2]
```diff
@@ -0,0 +1,668 @@
+From 616fa12ee39a8aedeba8d0f8b497d8959ee917b9 Mon Sep 17 00:00:00 2001
+From: Simo Piiroinen <simo.piiroinen@jolla.com>
+Date: Thu, 26 Nov 2020 19:29:29 +0200
+Subject: [PATCH] Implement SailfishOS specific private-privileged option
+
+Adds private-privileged=DIR profile / command line option. This is
+SailfishOs specific feature that allows one to limit which portions
```
[the remark this answers is not in the document]
Comment (spiiroin) [reply rpm/0007-Implement-SailfishOS-specific-private-privileg.patch:7]: should be Sailfish OS everywhere now

**Comment by spiiroin** [comment]
> I only found some typos in places that don't matter much. Is there any reason why this is still WIP?

Lack of testing & confidence. Turns out that testing then revealed that it had fundamental problems.

**Review by Tomin1 — APPROVED** [review]
LGTM. I tried this and it seems to work well now.

**Pull request merged**